node/how_to_with_npm/react.md

```diff
@@ -0,0 +1,250 @@
+surprising that it's the most requested framework when it comes to building web
+apps with Deno.
+
+This is a tutorial that walks you through building a simple React app with Deno
+in less than five minutes. The app will display a list of dinosaurs. When you
+click on one, it'll take you to a dinosaur page with more details.
+
+![demo of the app]()
+
+Here's the hosted version of the app. View the source here.
+
+## Create Vite Extra
+
+This tutorial will use [Vite](https://vitejs.dev/) to quickly scaffold a Deno
+and React app. Let's run:
+
+```sh
+deno run -A npm:create-vite-extra
+```
+
+We'll name our project "dinosaur-react-app". Then, `cd` into the newly created
+project folder.
+
+## Add a backend
+
+The next step is to add a backend API. We'll create a very simple API that
+returns information about dinosaurs.
+
+In the directory, let's create an `api` folder. In that folder, we'll create a
+`main.ts` file, which will run the server, and a `data.json`, which is the hard
+coded data.
+
+```
+mkdir api && touch api/data.json && touch api/main.ts
+```
+
+Copy and paste
+[this json file](https://github.com/denoland/deno-vue-example/blob/main/api/data.json)
+into your `api/data.json`.
+
+Then, let's update `api/main.ts`:
+
+```ts
+import { Application, Router } from "https://deno.land/x/oak/mod.ts";
+import { oakCors } from "https://deno.land/x/cors/mod.ts";
+import data from "./data.json" assert { type: "json" };
+
+const router = new Router();
+router
+  .get("/", (context) => {
+    context.response.body = "Welcome to dinosaur API!";
+  })
+  .get("/api", (context) => {
+    context.response.body = data;
+  })
+  .get("/api/:dinosaur", (context) => {
+    if (context?.params?.dinosaur) {
+      const filtered = data.filter((item) =>
+        item["name"].toLowerCase() === context.params.dinosaur.toLowerCase()
+      );
+      if (filtered.length === 0) {
+        context.response.body = "No dinosaurs found.";
+      } else {
+        context.response.body = filtered[0];
+      }
+    }
+  });
+
+const app = new Application();
+app.use(oakCors()); // Enable CORS for All Routes
+app.use(router.routes());
+app.use(router.allowedMethods());
+
+await app.listen({ port: 8000 });
+```
+
+This is a very simple API server using [`oak`](https://deno.land/x/oak) that
+will return dinosaur information based on the route. Let's start the API server:
+
+```
+deno run --allow-env --allow-net api/main.ts
+```
+
+If we go to `localhost:8000/api`, we see:
+
+![json response of dinosaurs](/static/api-response.png)
+
+Lookin' good so far.
+
+## Add a router
+
+Our app will have two routes: `/` and `/:dinosaur`.
+
+We'll use [`react-router-dom`](https://reactrouter.com/en/main) for our routing
+logic. Let's add that to our dependencies in `vite.config.mjs`:
+
+```mjs
+import { defineConfig } from "npm:vite";
+import react from "npm:@vitejs/plugin-react";
+
+import "npm:react";
+import "npm:react-dom/client";
+import "npm:react-router-dom"; // Add this line
```

Comment:
I made a PR to create-vite-extra to add version requirements for these https://github.com/bluwy/create-vite-extra/pull/13

Reply:
awesome, i can update these accordingly